quickstart/src/main/java/com/twilio/video/quickstart/activity/VideoActivity.java

```diff
@@ -270,35 +267,70 @@ private void addParticipant(Participant participant) {
         }
         participantIdentity = participant.getIdentity();
         videoStatusTextView.setText("Participant "+ participantIdentity + " joined");
+
         /*
-         * Stop rendering local video track in primary view and move it to thumbnail view
+         * Add participant renderer
          */
-        localVideoTrack.removeRenderer(primaryVideoView);
-        thumbnailVideoView.setVisibility(View.VISIBLE);
-        localVideoTrack.addRenderer(thumbnailVideoView);
-        localVideoView = thumbnailVideoView;
+        if (participant.getMedia().getVideoTracks().size() > 0) {
+            addParticipantVideo(participant.getMedia().getVideoTracks().get(0));
+        }
+
         /*
          * Start listening for participant media events
          */
         participant.getMedia().setListener(mediaListener());
     }
 
     /*
+     * Set primary view as renderer for participant video track
+     */
+    private void addParticipantVideo(VideoTrack videoTrack) {
+        moveLocalVideoToThumbnailView();
+        primaryVideoView.setMirror(false);
```

Comment:
Could there be a problem if moveLocalVideoToThumbnailView does nothing (because Visible == View.VISIBLE) and you still execute the other steps in this method?